File: lang/php/src/vac.fun/algorithms/sorting/SelectionSort.php
<?php
    namespace vac\fun\algorithms\sorting\SelectionSort;

    function sort(array &$arr): void {
        if (count($arr) == 0) {
            return;
        }

        $temp = NULL;

        $iMin = 0;

        for ($i = 0, $bound = count($arr) - 1; $i < $bound; $i++) {
            $iMin = $i;
            for ($j = $i + 1; $j <= $bound; $j++) {
                if ($arr[$j] < $arr[$iMin]) {
                    $iMin = $j;
                }
            }
            if ($iMin != $i) {
                $temp = $arr[$i];
                $arr[$i] = $arr[$iMin];
                $arr[$iMin] = $temp;
            }
        }
    }
?>

File: go/muse/algorithms/sorting/selection_sort/selection_sort.go
package selection_sort

func Sort(arr []int) {
	if len(arr) == 0 {
		return
	}

	for i := range len(arr) - 1 {
		iMin := i
		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[iMin] {
				iMin = j
			}
		}
		if iMin != i {
			arr[i], arr[iMin] = arr[iMin], arr[i]
		}
	}
}
